method/parse.go

```diff
@@ -91,3 +91,47 @@ func Parse(opts *ParseOpts) (*Definition, error) {
 
 	return methodDef, nil
 }
+
+type ParseConstructorOpts struct {
+	Obj types.Object
+}
+
+func ParseConstructor(opts *ParseConstructorOpts) (*Definition, error) {
```

Comment:
We should reuse Parse here, no need for the duplication, we can support everything Parse does.